dbcon/joblist/resourcemanager.h

```diff
@@ -402,7 +408,7 @@ class ResourceManager
     inline void returnMemory(int64_t amount, boost::shared_ptr<int64_t> sessionLimit)
     {
         atomicops::atomicAdd(&totalUmMemLimit, amount);
-        atomicops::atomicAdd(sessionLimit.get(), amount);
+        sessionLimit? atomicops::atomicAdd(sessionLimit.get(), amount): 0;
```

Comment:
It is harder to read and as I mentioned earlier it is better to introduce RM::returnMemory to avoid using shared_ptr.

Reply:
I added sessionLimit ? so I could pass NULL if I didn't want to pay attention to sessionLimit. 

The problem is that RM is a singleton, but SessionLimit is not. I'm not sure what RM::returnMemory would get you.